classpath/java/io/BufferedInputStream.java

```diff
@@ -54,28 +59,7 @@ public int read(byte[] b, int offset, int length) throws IOException {
 
       count += remaining;
       position += remaining;
-      offset += remaining;
-      length -= remaining;
     }
-
-    while (length > 0) {
```

Comment:
Without this loop, it seems like this function will never be able to read more that a single buffer's worth of data at a time.  Perhaps that's why you bumped the default size?

That seems less than idea to me.